tests/format/js/conditional/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -239,6 +273,680 @@ const testConsole = new TestConsole(
+  ) ?
+    "foo"
+  : "bar";
+
+// long conditional, short consequent/alternate, chained
+// (break on short consequents iff in chained ternary and its conditional broke)
+const longConditionalChained =
+  (
+    bifornCringerMoshedPerplexSawder === 2 / askTrovenaBeenaDependsRowans &&
+    glimseGlyphsHazardNoopsTieTie >=
+      averredBathersBoxroomBuggyNurl().anodyneCondosMalateOverateRetinol()
+  ) ?
+    "foo"
+  : anotherCondition ? "bar"
+  : "baz";
+
+// As a function parameter, don't add an extra indent:
+definition.encode(
+  typeof row[field] !== "undefined" ? row[field]
+  : typeof definition.default !== "undefined" ? definition.default
+  : null,
+  typeof row[field] === "undefined" ?
+    typeof definition.default === "undefined" ? null
+    : definition.default
+  : row[field]
+);
+
+// In a return, break and over-indent:
+const inReturn = () => {
+  if (short) {
+    return foo ? 1 : 2;
+  }
+  return typeof row[aVeryLongFieldName] !== "undefined" ?
+      row[aVeryLongFieldName]
+    : null;
+};
+
+// Remove current JSX Mode, and replace it with this algorithm:
+// When a ternary's parent is a JSXExpressionContainer which is not in a JSXAttribute,
+// force the consequent to break,
+// and if the alternate breaks,
+// add a newline before the closing curly brace.
+// Special case when the consequent is \`null\`:
+// do not add a line before or after it,
+// and wrap the alternate in parens.
+
+const someJSX = (
+  <div>
+    Typical jsx case:
+    {showFoo ?
+      <Foo attribute="such and such stuff here" />
+    : <Bar short />}
+    Nested, and with a non-jsx consequent is the same:
+    {component ?
+      React.createElement(component, props)
+    : render ?
+      <div>{render(props)}</div>
+    : <div>Nothing is here</div>}
+    As is a non-jsx consequent:
+    {showTheJSXElement ?
+      <div>the stuff</div>
+    : renderOtherStuff()}
+    But if the alternate breaks, add a newline before the closing curly brace:
+    {showTheThing || pleaseShowTheThing ?
+      <Foo attribute="such and such stuff here" />
+    : <Bar
+        attribute="such and such stuff here"
+        another="more stuff here"
+        third="and even more, hooray!"
+      />
+    }
+    When the consequent is \`null\` and the alternate breaks, hug it with parens
+    to match boolean behavior:
+    {!thing ? null : (
+      <TheThing
+        thing={thing}
+        someVeryLongPropertyThatBreaksTheAlternate="hello"
+      />
+    )}
+  </div>
+);
+
+ternaryWithJSXElements.hasNoSpecialCasing =
+  component ? <div>{React.createElement(component, props)}</div>
+  : render ? <div>{render(props)}</div>
+  : <div>Nothing is here</div>;
+
+jsxExpressionContainer.inJSXAttribute.hasNoSpecialCasing = (
+  <Foo
+    withJSX={isRed ? <RedColorThing /> : <GreenColorThing />}
+    withJSXBroken={
+      isRed || isSomeOtherLongCondition.thatBreaksTheLine() ?
+        <RedColorThing />
+      : <GreenColorThing />
+    }
+  />
+);
```

Comment:
Would be great to ship all those examples in master (regardless of this PR) and rebase so we can see the before/after your changes

Reply:
Done! 
PR here: https://github.com/prettier/prettier/pull/13952
Diff here: https://github.com/prettier/prettier/pull/13183/commits/8d99d9a91b112b9b783dcfbed708b9b91cdbb4eb